src/redux/userLists.js

```diff
@@ -0,0 +1,178 @@
+import produce from 'immer';
+import { concat, map, toLower, uniq, without } from 'lodash';
+import { InteractionManager } from 'react-native';
+import {
+  getSelectedUserList,
+  getUserLists,
+  saveSelectedUserList,
+  saveUserLists,
+} from '../handlers/localstorage/userLists';
+import { DefaultTokenLists } from '../references';
+import { emitAssetRequest } from './explorer';
+import { uniswapUpdateFavorites } from './uniswap';
+
+// -- Constants ------------------------------------------------------------- //
+const USER_LISTS_READY = 'userLists/USER_LISTS_READY';
+const USER_LISTS_LOAD_REQUEST = 'userLists/USER_LISTS_LOAD_REQUEST';
+const USER_LISTS_LOAD_SUCCESS = 'userLists/USER_LISTS_LOAD_SUCCESS';
+const USER_LISTS_LOAD_FAILURE = 'userLists/USER_LISTS_LOAD_FAILURE';
+
+const USER_LISTS_UPDATE_LISTS = 'userLists/USER_LISTS_UPDATE_LISTS';
+const USER_LISTS_CLEAR_STATE = 'userLists/USER_LISTS_CLEAR_STATE';
+const USER_LISTS_SET_SELECTED_LIST = 'userLists/USER_LISTS_SET_SELECTED_LIST';
+const FAVORITES_LIST_ID = 'favorites';
+// -- Actions --------------------------------------------------------------- //
+export const userListsLoadState = () => async (dispatch, getState) => {
+  const { network } = getState().settings;
+
+  dispatch({ type: USER_LISTS_LOAD_REQUEST });
+  try {
+    const defaultLists = DefaultTokenLists[network] || [];
+    const userLists = await getUserLists(network);
+    const lists = userLists?.length ? userLists : defaultLists;
+    let allAddresses = [];
+    lists.forEach(list => {
+      allAddresses = [...allAddresses, ...list.tokens];
+    });
+
+    dispatch({
+      payload: { lists },
+      type: USER_LISTS_LOAD_SUCCESS,
+    });
+    const selectedUserList = (await getSelectedUserList()) || FAVORITES_LIST_ID;
+    dispatch(userListsSetSelectedList(selectedUserList, false));
+
+    // Wait until the socket is ready
+    setTimeout(() => {
+      dispatch(emitAssetRequest(allAddresses));
+      dispatch({
+        payload: true,
+        type: USER_LISTS_READY,
+      });
+    }, 3000);
+  } catch (error) {
+    dispatch({ type: USER_LISTS_LOAD_FAILURE });
+  }
+};
+
+export const userListsResetState = () => dispatch =>
```

Comment:
can delete this function as it is no longer used and we don't expect to need to reset this